napari/layers/labels/labels.py

```diff
@@ -731,7 +801,11 @@ def paint(self, coord, new_label, refresh=True):
                 keep_coords = self.data[slice_coord] == self.selected_label
             else:
                 keep_coords = self.data[slice_coord] == self._background_label
-            self.data[slice_coord][keep_coords] = new_label
+            if self.brush_shape == "circle":
+                slice_coord = tuple(sc[keep_coords] for sc in slice_coord)
+                self.data[slice_coord] = new_label
+            else:
+                self.data[slice_coord][keep_coords] = new_label
```

Comment:
This new if/else is bugging me but after 30min of playing around with it in my head, I can't come up with something better, so I guess it stays for now! 😂